### PR TITLE
add edit office endpoint and fix find/create command

### PR DESCRIPTION
**Pull request by Cheephot**

### CODE REVIEW

File: src/CompanyService.Business/Commands/Office/FindOfficesCommand.cs [hunk 1]
```diff
@@ -15,47 +19,51 @@ public class FindOfficesCommand : IFindOfficesCommand
...
+        .Select(_mapper.Map)
+        .ToList();
+
+      response.TotalCount = totalCount;
+      response.Status = OperationResultStatusType.FullSuccess;
+
+      if (response.Body == null)
       {
-        _httpContextAccessor.HttpContext.Response.StatusCode = (int)HttpStatusCode.BadRequest;
+        _httpContextAccessor.HttpContext.Response.StatusCode = (int)HttpStatusCode.NotFound;
```
Comment (BelehovEgor) [inline src/CompanyService.Business/Commands/Office/FindOfficesCommand.cs:60]: we don't find nothing, but why it is error? why we don't return empty list

**Review by BelehovEgor — APPROVED** [review]

File: src/CompanyService.Business/Commands/Office/FindOfficesCommand.cs (hunk 1 above)
Comment (Cheephot) [reply src/CompanyService.Business/Commands/Office/FindOfficesCommand.cs:60]: done

File: src/CompanyService.Data/OfficeRepository.cs [hunk 2]
```diff
@@ -1,70 +1,86 @@
...
-        private readonly IDataProvider _provider;
+      _httpContextAccessor = httpContextAccessor;
+      _provider = provider;
+    }
 
-        public OfficeRepository(IDataProvider provider)
-        {
-            _provider = provider;
-        }
+    public void Add(DbOffice office)
```
Comment (VLiamin) [inline src/CompanyService.Data/OfficeRepository.cs:27]: Can you make this method bool? And then remove throw exception.

File: src/CompanyService.Data/OfficeRepository.cs [hunk 3]
```diff
@@ -1,70 +1,86 @@
...
+    public List<DbOffice> Find(BaseFindFilter filter, out int totalCount)
+    {
+      if (filter == null)
+      {
+        totalCount = 0;
+        return null;
+      }
+
+      IQueryable<DbOffice> dbOffices = _provider.Offices
+          .AsQueryable();
```
Comment (VLiamin) [inline src/CompanyService.Data/OfficeRepository.cs:52]: 4 spaces instead of 2.

**Review by VLiamin — APPROVED** [review]

File: src/CompanyService.Data/OfficeRepository.cs (hunk 2 above)
Comment (Cheephot) [reply src/CompanyService.Data/OfficeRepository.cs:27]: done

File: src/CompanyService.Data/OfficeRepository.cs (hunk 3 above)
Comment (Cheephot) [reply src/CompanyService.Data/OfficeRepository.cs:52]: done

**Review by borovHokage — APPROVED** [review]

**Review by MariyaSin — APPROVED** [review]

**Pull request merged**